tooling/helpers/rest_helper.py

```diff
@@ -18,28 +23,37 @@ def await_transcription_finish(id: str, api_key: str):
             if status["transcription_id"] != id:
                 continue
             if status["status"] not in ["in_progress", "in_query"]:
-                is_finished = True
+                return True
+        count+=1
+    print("Wait time for rest transcription was reached. This could be an error or your machine being slow (in the ladder case increase the limit)")
+    return False
 
 
-def transcribe_file_rest(filepath: str, api_key: str) -> str:
+def transcribe_file_rest(filepath: str, api_key: str, scale: str) -> RestResult:
+    result = RestResult(scale=scale)
+    start_time = time.time()
     r = None
     with open(filepath, "rb") as f:
         r = requests.post(
             "http://localhost:8393/transcriptions",
             files={"file": f},
             headers={"Authorization": api_key},
         )
-    if r.status_code != 200:
-        print(r.status_code)
-        print("Something went wrong")
-        return ""
+    if r.status_code != HTTPStatus.OK:
```

Comment:
checking for success but not for failure?

maybe try catch around all the requests.get...

Reply:
The only exception that can be thrown here is a connection error. 
And if the api is not reachable we want to cancel the benchmark anyway


See also https://requests.readthedocs.io/en/latest/user/quickstart/#errors-and-exceptions